[ADP-2878] remove remove wallet from dblayer

paolino opened the pull request:
This is part of the epic to remove (unused) multi-wallet support in DBLayer

- [x] Remove removeWallet from DBLayer record
- [x] Remove deleteWallet functionality from Wallet module

ADP-2878

paolino committed:
Remove removeWallet function from DBLayer.

paolino committed:
Remove deleteWallet functionality.

HeinrichApfelmus commented on lib/wallet/bench/restore-bench.hs:
```diff
@@ -831,8 +831,6 @@ bench_restoration
                         benchmarks proxy w wid0 wname0 benchname restorationTime
                     saveBenchmarkPoints benchname results
-                    forM_ wallets $ \(wid, _, _) ->
-                        unsafeRunExceptT (W.deleteWallet w wid)
```
👍  Each time that the function `withBenchDBLayer` is invoked, it creates and deletes a new temporary database file using `withSystemTempFile` — hence, calling `deleteWallet` at this point is superfluous, as deleting the database file will have the same effect.

HeinrichApfelmus reviewed (approved):
Looks good to me, thank you! 😊

paolino committed:
Remove impossible cover tags for database State Machine properties.

This tags have always 0 coverage because deleting the wallet is now impossible.
